core/templates/dev/head/domain/exploration/StateTopAnswerStatisticsFactory.js

```diff
@@ -0,0 +1,120 @@
+     * @property {number} frequency
+     * @property {boolean} isAddressed
+     */
+
+    /**
+     * TODO(brianrodri): Move this helper function into a proper service which
+     * takes the state's interaction type into account when formatting the HTML.
+     * @param {*} answer
+     * @returns {string}
+     */
+    var convertAnswerToHtml = function(answer) {
+      // Don't render quotes when the answer is a string.
+      return (typeof answer === 'string') ? answer : JSON.stringify(answer);
+    };
+
+    /**
+     * @private @constructor
+     * Returns a state's top answers with stale data. For a fresh instance, use
+     * {@link StateTopAnswerStatistics#createFromBackendDict} instead, or call
+     * {@link StateTopAnswerStatistics#refreshIsAddressed} after construction.
+     *
+     * @param {!string} stateName
+     * @param {!{answer, frequency: number}[]} backendTopAnswers
+     */
+    var StateTopAnswerStatistics = function(stateName, backendTopAnswers) {
+      /** @type {string} */
+      this._stateName = stateName;
+      /** @type {AnswerStatistics[]} */
+      this._answers = backendTopAnswers.map(function(answerFrequencyPair) {
+        return {
+          answer: angular.copy(answerFrequencyPair.answer),
+          answerAsHtml: convertAnswerToHtml(answerFrequencyPair.answer),
+          frequency: answerFrequencyPair.frequency,
+          isAddressed: false, // stale, needs to be updated.
+        };
+      });
+    };
+
+    /** Examines associated state to refresh the addressed info of answers. */
+    StateTopAnswerStatistics.prototype.refreshIsAddressed = function() {
+      var explorationId = ExplorationContextService.getExplorationId();
+      var state = ExplorationStatesService.getState(this._stateName);
+      var interactionRulesService = $injector.get(
+        AngularNameService.getNameOfInteractionRulesService(
+          state.interaction.id));
+
+      this._answers.forEach(function(answerStats) {
+        answerStats.isAddressed =
+          AnswerClassificationService.isClassifiedExplicitlyOrGoesToNewState(
+            explorationId, state.name, state, answerStats.answer,
+            interactionRulesService);
+      });
+    };
+
+    /** Returns rendering data for each answer and their statistics. */
+    StateTopAnswerStatistics.prototype.getAnswerStats = function() {
+      return this._answers.map(function(answerStats) {
+        return {
+          answer: answerStats.answerAsHtml,
+          frequency: answerStats.frequency,
+          isAddressed: answerStats.isAddressed,
+        };
+      });
+    };
+
+    /**
+     * Prepares a fresh new StateTopAnswerStatistics instance from backend data.
+     *
+     * This is the prefered way to create new instances because it ensures the
+     * data is fresh.
+     *
+     * @param {string} stateName - Name of the state this new instance will be
+     *     responsible for.
+     * @param {{answer, frequency: number}[]} backendDict
```

Comment:
docstring doesn't match function signature. Maybe also describe what's supposed to be in this array?